kube-runtime/src/controller/mod.rs

```diff
@@ -136,6 +155,61 @@ impl<T> Context<T> {
     }
 }
 
+/// A request to reconcile an object, annotated with why that request was made.
+///
+/// NOTE: The reason is ignored for comparison purposes. This means that, for example,
+/// an object can only occupy one scheduler slot, even if it has been scheduled for multiple reasons.
+/// In this case, only *the first* reason is stored.
+#[derive(Derivative)]
+#[derivative(
+    Debug(bound = "K::DynamicType: Debug"),
+    Clone(bound = "K::DynamicType: Clone"),
+    PartialEq(bound = "K::DynamicType: PartialEq"),
+    Eq(bound = "K::DynamicType: Eq"),
+    Hash(bound = "K::DynamicType: Hash")
+)]
+pub struct ReconcileRequest<K: Resource> {
+    pub obj_ref: ObjectRef<K>,
+    #[derivative(PartialEq = "ignore", Hash = "ignore")]
+    pub reason: ReconcileReason,
+}
```

Comment:
ah the core idea. i like it. smart way of injecting data without adding more to the applier.